### PR TITLE
DAOS-15352 rebuild: fix few bugs

### PR DESCRIPTION
**Pull request by wangshilong**
1. Fix to return real error rather than return ENOMEM which is very confusing.

2. skip not started pool when creating migrating pools.

4. skip up targets when updating cont prop.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by NiuYawei — DISMISSED** [review]

File: src/object/srv_obj_migrate.c [hunk 1]
```diff
@@ -637,7 +637,9 @@ migrate_pool_tls_lookup_create(struct ds_pool *pool, unsigned int version, unsig
 	arg.svc_list = (d_rank_list_t *)entry->dpe_val_ptr;
 	arg.obj_ult_cnts = tls->mpt_obj_ult_cnts;
 	arg.dkey_ult_cnts = tls->mpt_dkey_ult_cnts;
-	rc = dss_task_collective(migrate_pool_tls_create_one, &arg, 0);
+	rc = ds_pool_task_collective(pool->sp_uuid,
+				     PO_COMP_ST_NEW | PO_COMP_ST_DOWN | PO_COMP_ST_DOWNOUT,
+				     migrate_pool_tls_create_one, &arg, false);
```
Comment (liuxuezhao) [inline src/object/srv_obj_migrate.c:642]: I am not 100% sure if the "PO_COMP_ST_NEW | PO_COMP_ST_DOWN | PO_COMP_ST_DOWNOUT" tgt don't need the tls. Checked a while looks fine so far.

BTW, the last "false" better to be 0.

**Review by liuxuezhao — DISMISSED** [review]

File: src/container/srv_target.c [hunk 2]
```diff
@@ -2574,8 +2574,16 @@ ds_cont_tgt_prop_update(uuid_t pool_uuid, uuid_t cont_uuid, daos_prop_t	*prop)
 	uuid_copy(arg.cpa_cont_uuid, cont_uuid);
 	uuid_copy(arg.cpa_pool_uuid, pool_uuid);
 	arg.cpa_prop = prop;
+	/*
+	 * Before initiating the rebuild scan, the iv snap fetch function
+	 * will be invoked. This action may prompt a collective call to up targets
+	 * whose containers have not yet been created. Therefore, we should skip
+	 * the up targets in this scenario. The target property will be updated
+	 * upon initiating container aggregation.
+	 */
```
Comment (NiuYawei) [inline src/container/srv_target.c:2583]: The comment said it's about a snap IV fetch, so shouldn't we fix the collective call in ds_cont_tgt_snapshots_update()? A property update is also invoked on rebuild scan handler?

**Review by NiuYawei — CHANGES_REQUESTED** [review]

File: src/container/srv_target.c (hunk 2 above)
Comment (wangshilong) [reply src/container/srv_target.c:2583]: you are right, i think i miss-read the code when writting it.

**Review by NiuYawei — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by wangshilong** [comment]
> Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14097/10/execution/node/1481/log

known as DAOS-15431 rebuild_simple.

**Comment by mchaarawi** [comment]
I cannot force land this given that the failing test are rebuild related and the PR itself is also rebuild related. I will defer to another gatekeeper if they truly think this is OK and unrelated..

**Comment by wangshilong** [comment]
> I cannot force land this given that the failing test are rebuild related and the PR itself is also rebuild related. I will defer to another gatekeeper if they truly think this is OK and unrelated..

Let me try to rebase and test again.

**Pull request merged**